### ISSUE TITLE
tried to run the docker sample, get this error

### ISSUE THREAD

**Issue by GoodJoey** [uber/horovod#195]
run two containers in two nodes.
contrain1(10.240.209.120) and container2(10.240.208.206) can ssh to each other. 
i can run mpirun -np 2 -H 10.240.209.120:2 -mca plm_rsh_args "-p 12345"  
and mpirun -np 2 -H 10.240.208.206:2 -mca plm_rsh_args "-p 12345"  sucessfully (both run on contain1).

but can't run -np 4 -H 10.240.209.120:2,10.240.208.206:2 (use two nodes at the same time)

the progress gets stuck here: 

mpirun -np 4 -H 10.240.209.120:2,10.240.208.206:2 -mca plm_rsh_args "-p 12345" python keras_mnist_advanced.py
/usr/local/lib/python2.7/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
/usr/local/lib/python2.7/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
[train04][[9281,1],1][btl_tcp_endpoint.c:796:mca_btl_tcp_endpoint_complete_connect] connect() to 192.168.118.0 failed: No route to host (113)
[train01][[9281,1],0][btl_tcp_endpoint.c:796:mca_btl_tcp_endpoint_complete_connect] connect() to 172.19.0.1 failed: Connection timed out (110)

any help is welcomed and appreciated. Thanks.

**Comment by alsrgv** [comment]
Are you running docker with `--network=host`?

**Comment by GoodJoey** [comment]
on the contain1:
docker0   Link encap:Ethernet  HWaddr 02:42:dc:65:93:13  
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:dcff:fe65:9313/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:678285 errors:0 dropped:0 overruns:0 frame:0
          TX packets:944591 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:173139579 (173.1 MB)  TX bytes:2325496056 (2.3 GB)

enp0s31f6 Link encap:Ethernet  HWaddr 4c:cc:6a:b8:f7:ac  
          inet addr:10.240.209.120  Bcast:10.240.215.255  Mask:255.255.248.0
          inet6 addr: fe80::e8d7:2292:5203:431/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:176132821 errors:0 dropped:0 overruns:0 frame:0
          TX packets:119234624 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:137104274061 (137.1 GB)  TX bytes:44956972589 (44.9 GB)
          Interrupt:16 Memory:db100000-db120000 

enp7s0    Link encap:Ethernet  HWaddr 4c:cc:6a:b8:f7:ad  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
          Memory:df500000-df51ffff 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:94779517 errors:0 dropped:0 overruns:0 frame:0
          TX packets:94779517 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:47701643629 (47.7 GB)  TX bytes:47701643629 (47.7 GB)

tunl0     Link encap:IPIP Tunnel  HWaddr   
          inet addr:192.168.118.0  Mask:255.255.255.255
          UP RUNNING NOARP  MTU:1440  Metric:1
          RX packets:1427887 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1017535 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:6266604377 (6.2 GB)  TX bytes:7278786270 (7.2 GB)

**Comment by alsrgv** [comment]
Using the instructions from https://github.com/uber/horovod/blob/master/docs/running.md#hangs-due-to-non-routed-network-interfaces, you can disable all the interfaces you don't need.

`-mca btl_tcp_if_exclude docker0,enp7s0,lo,tunl0 
 -x NCCL_SOCKET_IFNAME=^docker0,^enp7s0,^lo,^tunl0`

I'm not sure if `NCCL_SOCKET_IFNAME` needs just one ^ in the beginning or before each interface - you can try both options.

**Comment by GoodJoey** [comment]
worked, thanks. @alsrgv
